sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -309,19 +314,88 @@ private PlannerResult planWithBindableConvention(
     } else {
       final BindableRel theRel = bindableRel;
       final DataContext dataContext = plannerContext.createDataContext((JavaTypeFactory) planner.getTypeFactory());
-      final Supplier<Sequence<Object[]>> resultsSupplier = new Supplier<Sequence<Object[]>>()
-      {
-        @Override
-        public Sequence<Object[]> get()
-        {
-          final Enumerable enumerable = theRel.bind(dataContext);
-          return Sequences.simple(enumerable);
-        }
-      };
+
+      final Supplier<Sequence<Object[]>> resultsSupplier = () -> new BaseSequence<>(
+          new BaseSequence.IteratorMaker<Object[], CloseableEnumerableIterator>()
+          {
+            @Override
+            public CloseableEnumerableIterator make()
+            {
+              final Enumerable enumerable = theRel.bind(dataContext);
+              final Enumerator enumerator = enumerable.enumerator();
+              return new CloseableEnumerableIterator(new Iterator<Object[]>()
```

Comment:
It was needed, since we found that earlier the iterator was not being closed which could cause a potential leak. I need to better understand, how does `Sequences.withBaggage()` works.

Reply:
I'm not saying closing the iterator is not needed. The same algorithm can be implemented using `Sequences.withBaggage()` , so `CloseableEnumerableIterator` itself is not needed. `Sequences.withBaggage()` wraps the underlying sequence and calls the given `closeable` after iterating all elements in the sequence.

Reply:
okay, will use `Sequences.withBaggage()`, but it seems i still need to override the `make` from `BaseSequence`. Changed to use `Sequences.withBaggage()` with an `EnumeratorIterator`.

Reply:
Thanks.